litestar/serialization/msgspec_hooks.py

```diff
@@ -115,6 +118,12 @@ def default_deserializer(
     if issubclass(target_type, (Path, PurePath, ImmutableState, UUID)):
         return target_type(value)
 
+    if issubclass(target_type, SecretBytes) and isinstance(value, (bytes, str)):
```

Comment:
Not a review, but I recently found out that you can do `bytes | str` instead of `(bytes, str)`, but I'm not sure from which Python version this is supported. I thought that was pretty cool.

Reply:
Interesting!